ibis/pyspark/compiler.py

```diff
@@ -1111,53 +1092,69 @@ def compile_window_op(t, expr, scope, timecontext, **kwargs):
         else:
             pyspark_window = pyspark_window.rowsBetween(start, end)
 
-    result = t.translate(
-        operand, scope, timecontext, window=pyspark_window, context=context
-    )
-    return result
+    res_op = operand.op()
+    if isinstance(res_op, (ops.NotAll, ops.NotAny)):
+        # For NotAll and NotAny, negation must be applied after .over(window)
+        # Here we rewrite node to be its negation, and negate it back after
+        # translation and window operation
+        return ~(
+            t.translate(
+                res_op.negate().to_expr(), scope, timecontext, context=context
+            ).over(pyspark_window)
+        )
+    elif isinstance(res_op, (ops.MinRank, ops.DenseRank, ops.RowNumber)):
+        # result must be cast to long type for rank / rownumber
+        return (
+            t.translate(operand, scope, timecontext, context=context)
```

Comment:
factor out this to a variable
` t.translate(operand, scope, timecontext, context=context)`